src/pages/index.js

```diff
@@ -60,40 +99,200 @@ const cardList = new Section(
+      addNewCard.close();
+      addCardForm.reset();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
 
+    .finally(() => {
+      console.log("Post success.");
+      addNewCard.setLoading(false);
+    });
+}
+
+function createCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageAction,
+    handleDeleteCard,
+    likeCard,
+    unlikeCard
+  );
+  return card.getView();
+}
+
+api
+  .getInitialCards()
+  .then((res) => {
+    console.log(res);
+    cardList.renderItems(res);
+  })
+
+  .catch((err) => alert(err));
+
+/**************************************************************************
+ *                               Like                                      *
+ **************************************************************************/
+
+function likeCard(card) {
+  api
+    .likeCard(card.id, card.isLiked)
+    .then((res) => {
+      console.log(res);
+      card.setIsLiked(res.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      console.log("Card Liked");
+    });
+}
+/**************************************************************************
+ *                               Unlike                                    *
+ **************************************************************************/
+
+function unlikeCard(card) {
+  api
+    .unlikeCard(card.id)
+    .then((res) => {
+      console.log(res);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+/**************************************************************************
+ *                               addNewCard                               *
+ **************************************************************************/
 const addNewCard = new PopupWithForm(
   "#profile-add-modal",
   handleAddCardFormSubmit
 );
 addNewCard.setEventListeners();
 
-//addCard .................................
+/**************************************************************************
+ *                               User Info                                *
+ **************************************************************************/
+
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__avatar"
+);
 
-//edit profile .................................
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUsersInfo(res);
+    userInfo.updateAvaImg(res);
+  })
+  .catch((err) => alert(err));
+
+/**************************************************************************
+ *                               Edit Profile                             *
+ **************************************************************************/
 
 const editProfileModal = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 editProfileModal.setEventListeners();
 
-//edit profile .................................
+function handleProfileEditSubmit(formValues) {
+  editProfileModal.setLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```